src/RA_Dlg_MemBookmark.cpp

```diff
@@ -20,437 +22,434 @@ namespace ra {
                 }
-                case IDC_RA_DECIMALBOOKMARK:
+
+                return TRUE;
+            }
+            case IDC_RA_DECIMALBOOKMARK:
+            {
+                if (!m_vBookmarks.empty())
                 {
-                    if (m_vBookmarks.size() > 0)
+                    auto hList = GetDlgItem(hDlg, IDC_RA_LBX_ADDRESSES);
+                    const auto uSelectedCount = ListView_GetSelectedCount(hList);
+
+                    if (uSelectedCount > 0)
                     {
-                        hList = GetDlgItem(hDlg, IDC_RA_LBX_ADDRESSES);
-                        const unsigned int uSelectedCount = ListView_GetSelectedCount(hList);
-
-                        if (uSelectedCount > 0)
-                        {
-                            for (int i = ListView_GetNextItem(hList, -1, LVNI_SELECTED); i >= 0; i = ListView_GetNextItem(hList, i, LVNI_SELECTED))
-                                m_vBookmarks[i]->SetDecimal(!m_vBookmarks[i]->Decimal());
-                        }
-                        ListView_SetItemState(hList, -1, LVIF_STATE, LVIS_SELECTED);
+                        for (auto i = ListView_GetNextItem(hList, -1, LVNI_SELECTED); i >= 0;
+                             i = ListView_GetNextItem(hList, i, LVNI_SELECTED))
+                            m_vBookmarks.at(i).SetDecimal(!m_vBookmarks.at(i).Decimal());
                     }
-                    return TRUE;
-                }
-                case IDC_RA_SAVEBOOKMARK:
-                {
-                    ExportJSON();
-                    return TRUE;
+                    ListView_SetItemState(hList, -1, LVIF_STATE, LVIS_SELECTED);
                 }
-                case IDC_RA_LOADBOOKMARK:
-                {
-                    std::wstring file = ImportDialog();
-                    if (file.length() > 0)
-                        ImportFromFile(file);
-                    return TRUE;
-                }
-                default:
-                    return FALSE;
+                return TRUE;
             }
+            case IDC_RA_SAVEBOOKMARK:
+            {
+                ExportJSON();
+                return TRUE;
+            }
+            case IDC_RA_LOADBOOKMARK:
+            {
+                std::wstring file{ImportDialog()};
+                if (!file.empty())
+                    ImportFromFile(std::move(file));
+                return TRUE;
+            }
+            default:
+                return FALSE;
         }
+    };
+
+    switch (uMsg)
+    {
+        HANDLE_MSG(hDlg, WM_INITDIALOG, OnInitDialog);
+        HANDLE_MSG(hDlg, WM_GETMINMAXINFO, OnGetMinMaxInfo);
+        HANDLE_MSG(hDlg, WM_SIZE, OnSize);
+        HANDLE_MSG(hDlg, WM_MEASUREITEM, OnMeasureItem);
+        HANDLE_MSG(hDlg, WM_DRAWITEM, OnDrawItem);
+        HANDLE_MSG(hDlg, WM_NOTIFY, OnNotify);
+        HANDLE_MSG(hDlg, WM_COMMAND, OnCommand);
+
+        case WM_MOVE:
+            RememberWindowPosition(hDlg, "Memory Bookmarks");
+            break;
```

Comment:
The stuff in `WM_MOVE` seemed like it didn't need a callback for the message cracker.